lib/resourceapply/cv.go

```diff
@@ -1,107 +1,81 @@
-	osv1 "github.com/openshift/cluster-version-operator/pkg/apis/operatorstatus.openshift.io/v1"
-	cvclientv1 "github.com/openshift/cluster-version-operator/pkg/generated/clientset/versioned/typed/clusterversion.openshift.io/v1"
-	osclientv1 "github.com/openshift/cluster-version-operator/pkg/generated/clientset/versioned/typed/operatorstatus.openshift.io/v1"
-	cvlistersv1 "github.com/openshift/cluster-version-operator/pkg/generated/listers/clusterversion.openshift.io/v1"
-	oslistersv1 "github.com/openshift/cluster-version-operator/pkg/generated/listers/operatorstatus.openshift.io/v1"
+	cvv1 "github.com/openshift/cluster-version-operator/pkg/apis/config.openshift.io/v1"
+	cvclientv1 "github.com/openshift/cluster-version-operator/pkg/generated/clientset/versioned/typed/config.openshift.io/v1"
+	cvlistersv1 "github.com/openshift/cluster-version-operator/pkg/generated/listers/config.openshift.io/v1"
 	"k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/utils/pointer"
 )
 
-func ApplyOperatorStatus(client osclientv1.ClusterOperatorsGetter, required *osv1.ClusterOperator) (*osv1.ClusterOperator, bool, error) {
-	if required.Status.Extension.Raw != nil && required.Status.Extension.Object != nil {
-		return nil, false, fmt.Errorf("both extension.Raw and extension.Object should not be set")
-	}
-	existing, err := client.ClusterOperators(required.Namespace).Get(required.Name, metav1.GetOptions{})
+func ApplyClusterVersion(client cvclientv1.ClusterVersionsGetter, required *cvv1.ClusterVersion) (*cvv1.ClusterVersion, bool, error) {
+	existing, err := client.ClusterVersions().Get(required.Name, metav1.GetOptions{})
 	if errors.IsNotFound(err) {
-		actual, err := client.ClusterOperators(required.Namespace).Create(required)
+		actual, err := client.ClusterVersions().Create(required)
 		return actual, true, err
 	}
 	if err != nil {
 		return nil, false, err
 	}
 
 	modified := pointer.BoolPtr(false)
-	resourcemerge.EnsureOperatorStatus(modified, existing, *required)
+	resourcemerge.EnsureClusterVersion(modified, existing, *required)
 	if !*modified {
 		return existing, false, nil
 	}
 
-	actual, err := client.ClusterOperators(required.Namespace).Update(existing)
+	actual, err := client.ClusterVersions().Update(existing)
 	return actual, true, err
 }
 
-func ApplyOperatorStatusFromCache(lister oslistersv1.ClusterOperatorLister, client osclientv1.ClusterOperatorsGetter, required *osv1.ClusterOperator) (*osv1.ClusterOperator, bool, error) {
-	if required.Status.Extension.Raw != nil && required.Status.Extension.Object != nil {
-		return nil, false, fmt.Errorf("both extension.Raw and extension.Object should not be set")
-	}
-	existing, err := lister.ClusterOperators(required.Namespace).Get(required.Name)
+func ApplyClusterVersionFromCache(lister cvlistersv1.ClusterVersionLister, client cvclientv1.ClusterVersionsGetter, required *cvv1.ClusterVersion) (*cvv1.ClusterVersion, bool, error) {
```

Comment:
@abhinavdahiya as a general statement, is there a reason I can't shim a getter to use a lister for `get` and avoid this?

@mfojtik another mike or Luis?

Reply:
> @abhinavdahiya as a general statement, is there a reason I can't shim a getter to use a lister for get and avoid this?

@deads2k can you expand on it a little bit.. :innocent:

Reply:
The point of this `FromCache` appears to be to avoid `GET` pressure on the server.  That makes sense, since a cache is already available, but instead of having a separate method, it seems like a clever impl could do the trick with an existing method.  We'll probably knock something together next week (master team operators are GET happy)

Reply:
Awesome, can you point me to an example in master operators